multimetric/cls/calc/maintenance.py

```diff
@@ -176,11 +185,44 @@ def _mi_classic(metrics):
             return max(0, res)
         except ValueError:  # pragma: no cover
             return 0  # pragma: no cover
+        
+    @staticmethod
+    def _mi_individual(metrics,formula):
+        """Calculate the maintainability index using a given method by the user.
+
+        Parameters
+        ----------
+        metrics : dict
+            The metrics used for calculation.
+
+        formula : string
+            The formula which is used to evaluate the maintainability index
+
+        Returns
+        -------
+        float
+            The calculated maintenance index.
+
+        Raises
+        ------
+        ValueError
+            If any of the required metrics are missing.
+
+        Notes
+        -----
+
+        """
+        try:
+            local_namespace = {'math': math}
+            return eval(formula,local_namespace,metrics)
```

Comment:
:grimacing: that is bad, as one could define arbitrary commands to be executed through the tool.
I see the use case, but we need a way to configure that through a static format, one that can be validated before being run.

Another way could be allow plugins for the tool, so this class could be forked and tuned according to your needs.

Not sure what the best way is yet, but I would personally favor a plugin architecture

Reply:
Okay understandable. My main language is C/C++ and I use python for some tooling in a basic form. I thought with the locale namespace thing and the given variable, that execution is save as this would be the only things one might have access to.